src/components/PopupWithForm.js

```diff
@@ -1,11 +1,14 @@
 import Popup from "./Popup.js";
 
 class PopupWithForm extends Popup {
-  constructor(popupSelector, formSelector, handleFormSubmit) {
+  constructor(popupSelector, formSelector, handleFormSubmit, submitText) {
     super({ popupSelector });
     this._form = document.forms[formSelector];
     this._handleFormSubmit = handleFormSubmit;
     this._inputList = this._form.querySelectorAll("input");
+    this._submitBtn = this._form.querySelector(".form__submit");
+    this._submitText = submitText;
+    this._setButtonText();
```

Comment:
I don't have initial textContent. I removed it from html, because I think it's good not to have all the popups in the DOM all the time. I want to create them only if necessary in the future and I prepare them to it. So I think the button text may be one of form variables in the future.